// @tailus/themer/dist/config/preconfigs/padding/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.largest = exports.larger = exports.large = exports.medium = exports.normal = exports.small = void 0;
var small_1 = __importDefault(require("./small"));
exports.small = small_1.default;
var normal_1 = __importDefault(require("./normal"));
exports.normal = normal_1.default;
var medium_1 = __importDefault(require("./medium"));
exports.medium = medium_1.default;
var large_1 = __importDefault(require("./large"));
exports.large = large_1.default;
var larger_1 = __importDefault(require("./larger"));
exports.larger = larger_1.default;
var largest_1 = __importDefault(require("./largest"));
exports.largest = largest_1.default;
exports.default = {
    small: small_1.default,
    normal: normal_1.default,
    medium: medium_1.default,
    large: large_1.default,
    larger: larger_1.default,
    largest: largest_1.default,
};
