// @tailus/themer/dist/config/preconfigs/palette/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.oz = exports.spring = exports.winter = exports.energy = exports.romance = exports.mystery = exports.trust = exports.nature = exports.passion = void 0;
var defaultFeedbackColors_1 = require("./defaultFeedbackColors");
exports.passion = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "red", secondary: "orange", accent: "amber", gray: "neutral" });
exports.nature = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "green", secondary: "emerald", accent: "teal", gray: "greenish", success: "lime" });
exports.trust = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "blue", secondary: "indigo", accent: "violet", gray: "slate" });
exports.mystery = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "purple", secondary: "fuchsia", accent: "pink", gray: "gray" });
exports.romance = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "rose", secondary: "red", accent: "orange", gray: "zinc" });
exports.energy = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "amber", secondary: "yellow", accent: "lime", gray: "stone" });
exports.winter = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "cyan", secondary: "sky", accent: "emerald", gray: "cyanish" });
exports.spring = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "lime", secondary: "green", accent: "teal", gray: "greenish" });
exports.oz = __assign(__assign({}, defaultFeedbackColors_1.feedbackColors), { primary: "indigo", secondary: "cyan", accent: "fuchsia", gray: "indigray", success: "lime", danger: "rose" });
